packages/starknet-snap/src/state/network-state-manager.test.ts

```diff
@@ -0,0 +1,243 @@
+import { constants } from 'starknet';
+
+import type { Network } from '../types/snapState';
+import {
+  STARKNET_MAINNET_NETWORK,
+  STARKNET_SEPOLIA_TESTNET_NETWORK,
+} from '../utils/constants';
+import { mockState } from './__tests__/helper';
+import { NetworkStateManager, ChainIdFilter } from './network-state-manager';
+import { StateManagerError } from './state-manager';
+
+describe('NetworkStateManager', () => {
+  describe('findNetwork', () => {
+    it('returns the token', async () => {
+      const chainId = constants.StarknetChainId.SN_SEPOLIA;
+      await mockState({
+        networks: [STARKNET_MAINNET_NETWORK, STARKNET_SEPOLIA_TESTNET_NETWORK],
+      });
+
+      const stateManager = new NetworkStateManager();
+      const result = await stateManager.findNetwork({
+        chainId,
+      });
+
+      expect(result).toStrictEqual(STARKNET_SEPOLIA_TESTNET_NETWORK);
+    });
+
+    it('returns null if the token can not be found', async () => {
```

Comment:
replace "returns the token" by "returns the chainId" ?

Reply:
returns the network